### PR TITLE
Don't rely on webpack polyfilling path

### LINKED ISSUE #3667: Map DDS has dependency on "path", which doesn't exist in the browser
## Describe the bug
The map DDS has a dependency on the path node module in directory.ts, which forces a polyfill to be included by webpack for path. Can we remove or replace this dependency with something that is browser compatible? 

### To Reproduce
1. Bundle the Map DDS in webpack and set `node: { path: false}` in the webpack config
2. Webpack emits an error of the following form "Can't resolve path'

### Expected behavior
It should bundle in Wepack 5

### PR DESCRIPTION
**Pull request by christiango**
Resolves #3667, where we are seeing issues in Webpack 5 due to the reliance on directory.ts on the bundler polyfilling the node Path usage here. 

Ideally, we'd use a lean utility library to implement this functionality to minimize bundle size, but as a step to unblock webpack we will explicitly require path-browserify

### CODE REVIEW

**Review by tylerbutler — APPROVED** [review]
🙏

**Comment by christiango** [comment]
@ChumpChief  - any concerns with this PR or can we merge?

File: packages/dds/map/src/directory.ts [hunk 1]
```diff
@@ -40,12 +39,11 @@ import {
...
 
-// path-browserify only supports posix functionality but doesn't have a path.posix to enforce it.  But we need to
-// enforce posix when using the normal node module on Windows (otherwise it will use path.win32).  Also including an
-// assert here to help protect in case path-browserify changes in the future, because we only want posix path
-// functionality.
-const posix = path.posix || path;
-assert(posix.sep === "/");
+// We use path-browserify since this code can run safely on the server or the browser.
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const path: typeof import("path") = require("path-browserify");
```
Comment (vladsud) [inline packages/dds/map/src/directory.ts:44]: Can / should this be an import instead of require?

**Review by vladsud — APPROVED** [review]

**Comment by ChumpChief** [comment]
+1 to Vlad's suggestion of import rather than require but otherwise seems good, thank you!

**Review by ChumpChief — APPROVED** [review]

**Pull request merged**